sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/AnsiTypeCoercion.scala

```diff
@@ -120,6 +120,9 @@ object AnsiTypeCoercion extends TypeCoercionBase {
     case (_: TimestampType, _: DateType) | (_: DateType, _: TimestampType) =>
       Some(TimestampType)
 
+    case (_: DayTimeIntervalType, _: DayTimeIntervalType) => Some(DayTimeIntervalType())
```

Comment:
shall we pick the smaller start field and larger end field as the result fields?